### PR TITLE
Dlg factor data frame

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/instat  this is ready for review

### CODE REVIEW

**Comment by Lunalo** [comment]
image
I think we need to clear the ucrInput whenever the dataframe is changed

**Comment by lilyclements** [comment]
I'll sort the ucrInput, as for the alignment - I think we all need to have a discussion over this. Some dialogs have it aligned at the top and some have it lower by a few spaces. I'm not sure what we want in general, but we do need consistency

**Comment by lilyclements** [comment]
I don't think we need to clear it. The user might put in a name they want, then decide they wanted to name another column this. If they clear the receiver to put the new column in, they may want to keep the name they put in?

**Comment by dannyparsons** [comment]
If the user has typed their own name in, then it shouldn't clear, but if they have not typed in something then I think it should update as the receiver contents changes, which I think is how it's implemented at the moment.

**Comment by lilyclements** [comment]
Yep, so currently:
Put in Column A
New Name is automatically A1
Change InputNewName to a name you wish to call it (e.g. "MyName")
Put Column B in, the InputNewName remains as "MyName"

If you didn't change the name from the automatic A1, then put in column B, it automatically changes the new column name to B1

**Comment by lilyclements** [comment]
Yep, so currently thats what it does:
Put in Column A
New Name is automatically A1
Change InputNewName to a name you wish to call it (e.g. "MyName")
Put Column B in, the InputNewName remains as "MyName"

If you didn't change the name from the automatic A1, then put in column B, it automatically updates the new column name to B1

**Comment by lilyclements** [comment]
This code has been re-updated since the updated dialog setup and added conditions in #2618 so this is ready for review.

**Comment by Lunalo** [comment]
Generally, the dialog is working fine.

File: instat/dlgFactorDataFrame.vb [hunk 1]
```diff
@@ -61,66 +93,21 @@ Public Class dlgFactorDataFrame
...
     End Sub
 
-    Private Sub FactorDataFrameName()
-        If Not ucrInputFactorNames.IsEmpty Then
-            ucrBase.clsRsyntax.AddParameter("factor_data_frame_name", Chr(34) & ucrInputFactorNames.GetText & Chr(34))
-        Else
-            ucrBase.clsRsyntax.RemoveParameter("factor_data_frame_name")
+    Private Sub CheckAutoName()
+        If Not ucrReceiverFactorDataFrame.IsEmpty AndAlso Not ucrInputFactorNames.bUserTyped Then
+            ucrInputFactorNames.SetName(ucrReceiverFactorDataFrame.GetVariableNames(False) & "1")
```
Comment (dannyparsons) [inline instat/dlgFactorDataFrame.vb]: Not sure why you want to add "1" to the column name as the default. Just the column name seems like a sensible default.

File: instat/dlgFactorDataFrame.vb [hunk 2]
```diff
@@ -17,42 +17,74 @@
...
     Private Sub ReopenDialog()
     End Sub
 
+    Private Sub SetRCodeforControls(bReset As Boolean)
+        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
+    End Sub
+
     Private Sub TestOKEnabled()
-        If Not ucrReceiverFactorDataFrame.IsEmpty AndAlso Not ucrInputFactorNames.IsEmpty Then
+        If Not ucrReceiverFactorDataFrame.IsEmpty AndAlso Not ucrInputFactorNames.IsEmpty AndAlso (ucrReceiverFactorDataFrame.GetVariableNames(bWithQuotes:=False) <> ucrInputFactorNames.GetText) Then ' and also receiver dataframe is empty
```
Comment (dannyparsons) [inline instat/dlgFactorDataFrame.vb]: I don't see why the new name cannot be the name of the selected column?

**Comment by lilyclements** [comment]
@dannyparsons I've made the necessary changes

**Pull request merged**